runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -203,7 +288,14 @@ function M.on_publish_diagnostics(_, result, ctx, config)
     vim.diagnostic.config(config, namespace)
   end
 
-  vim.diagnostic.set(namespace, bufnr, diagnostic_lsp_to_vim(diagnostics, bufnr, client_id))
+  -- Do not report diagnostics with tags as a normal diagnostics.
+  -- Instead, just highlight them. See:
+  -- https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#diagnosticTag
+  -- and https://github.com/microsoft/pyright/issues/1118#issuecomment-835528161
+  local diags, tagged_diags = filter_tagged(diagnostics)
+
+  vim.diagnostic.set(namespace, bufnr, diagnostic_lsp_to_vim(diags, bufnr, client_id))
+  highlight_tagged_diagnostics(bufnr, client_id, namespace_aux, tagged_diags)
```

Comment:
Do you think it's worth doing the conversion and filtering in one pass, to avoid a copy operation?

Not sure about the performance impact, but some language servers on bigger project can easily report 1000+ diagnostics.